client/src/contexts/UserContext.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { createContext, useContext } from 'react';
+import { UserType } from '../utils/types';
+import { useState } from 'react';
+
+const UserContext = createContext<UserType>({
+    username: "default",
+    pfp: "null"
+});
+
+const [user, setUser] = useState<UserType>({
+    username: "default",
+    pfp: "null"
+});
+
+export const useUser = () => {
+    return useContext(UserContext);
+}
+
+export const changeUsername = (newUsername: string) => {
+    setUser
+}
+
```

Comment:
could you implement a useEffect to randomly generate a username on first render? and then make sure it is not re-generated again?

use the generateUsername function found in ulities (or maybe in deprecated im not sure)